src/DwitTech.AccountService.Core/Services/ActivationService.cs

```diff
@@ -1,24 +1,26 @@
 ﻿using DwitTech.AccountService.Core.Interfaces;
 using DwitTech.AccountService.Core.Utilities;
-using Microsoft.AspNetCore.Http;
+using DwitTech.AccountService.Data.Repository;
 using Microsoft.Extensions.Configuration;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net;
-using System.Net.Mail;
-using System.Text;
-using System.Threading.Tasks;
 
 namespace DwitTech.AccountService.Core.Services
 {
     public class ActivationService : IActivationService
     {
-        private readonly IConfiguration _configuration; //Config instance for GetBaseUrl method
+        private readonly IConfiguration _configuration;
+        private readonly IUserRepository _userRepository;
+        private readonly IEmailService _emailService;
 
-        public ActivationService(IConfiguration configuration)
+
+        public ActivationService(IConfiguration configuration, IUserRepository userRepository, IEmailService emailService)
         {
             _configuration = configuration;
+            _userRepository = userRepository;
+            _emailService = emailService;
+        }
+
+        public ActivationService(IConfigurationRoot configuration, IUserRepository @object)
```

Comment:
Please explain the purpose of this constructor. I'd like to learn.

Reply:
i don't know to explain, but maybe when you have time we can get on a call.